kyaml/yaml/datamap.go

```diff
@@ -95,6 +95,35 @@ func makeSecretValueRNode(s string) *RNode {
 	return NewRNode(yN)
 }
 
+func (rn *RNode) LoadMapIntoSecretStringData(m map[string]string) error {
+	for _, k := range SortedMapKeys(m) {
+		fldName, vrN := makeSecretStringValueRNode(m[k])
+		if _, err := rn.Pipe(
+			LookupCreate(MappingNode, fldName),
+			SetField(k, vrN)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// valid UTF-8 strings can be stringData, but fallback to Base64 for non UTF-8
```

Comment:
Note that the documentation capitalizes "Base64" inconsistently, often writing it as all-lowercase.
```suggestion
// makeSecretStringValueRNode creates a scalar node for one of two potential fields.
// If the supplied node's value is a valid UTF-8 string, it use the "stringData" field, but
// otherwise falls back to placing the Base64-encoded value in the "data" field for
// non-UTF-8 strings.
```